File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcRelationship/IfcRelConnects/IfcRelSpaceBoundary/IfcRelSpaceBoundary.java
public class IfcRelSpaceBoundary extends IfcRelConnects {
    public IfcSpaceBoundarySelect RelatingSpace;
    public IfcElement RelatedBuildingElement;
    public IfcConnectionGeometry ConnectionGeometry;
    public IfcPhysicalOrVirtualEnum PhysicalOrVirtualBoundary;
    public IfcInternalOrExternalEnum InternalOrExternalBoundary;

    // === EXTENDED BY ===
    // IfcRelSpaceBoundary1stLevel

    // === WHERE CLAUSES ===
    // CorrectPhysOrVirt : ((PhysicalOrVirtualBoundary = IfcPhysicalOrVirtualEnum.Physical) AND (NOT('IFC4X3_DEV_73740fe4.IFCVIRTUALELEMENT' IN TYPEOF(RelatedBuildingElement)))) OR ((PhysicalOrVirtualBoundary = IfcPhysicalOrVirtualEnum.Virtual) AND (('IFC4X3_DEV_73740fe4.IFCVIRTUALELEMENT' IN TYPEOF(RelatedBuildingElement)) OR   ('IFC4X3_DEV_73740fe4.IFCOPENINGELEMENT' IN TYPEOF(RelatedBuildingElement)))) OR (PhysicalOrVirtualBoundary = IfcPhysicalOrVirtualEnum.NotDefined)
}

File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcRelationship/IfcRelConnects/IfcRelSpaceBoundary/IfcRelSpaceBoundary.js
class IfcRelSpaceBoundary extends IfcRelConnects {
    constructor() {
        /** @type {IFCSPACEBOUNDARYSELECT} */
        this.RelatingSpace = null;
        /** @type {IFCELEMENT} */
        this.RelatedBuildingElement = null;
        /** @type {IFCCONNECTIONGEOMETRY} */
        this.ConnectionGeometry = null;
        /** @type {IFCPHYSICALORVIRTUALENUM} */
        this.PhysicalOrVirtualBoundary = null;
        /** @type {IFCINTERNALOREXTERNALENUM} */
        this.InternalOrExternalBoundary = null;
    }

    // === EXTENDED BY ===
    // IfcRelSpaceBoundary1stLevel

    // === WHERE CLAUSES ===
    // CorrectPhysOrVirt : ((PhysicalOrVirtualBoundary = IfcPhysicalOrVirtualEnum.Physical) AND (NOT('IFC4X3_DEV_73740fe4.IFCVIRTUALELEMENT' IN TYPEOF(RelatedBuildingElement)))) OR ((PhysicalOrVirtualBoundary = IfcPhysicalOrVirtualEnum.Virtual) AND (('IFC4X3_DEV_73740fe4.IFCVIRTUALELEMENT' IN TYPEOF(RelatedBuildingElement)) OR   ('IFC4X3_DEV_73740fe4.IFCOPENINGELEMENT' IN TYPEOF(RelatedBuildingElement)))) OR (PhysicalOrVirtualBoundary = IfcPhysicalOrVirtualEnum.NotDefined)
}
